### PR TITLE
docs: update customize theme docs

### PR DESCRIPTION
**Pull request by AshoneA**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/issues/23154#issuecomment-612569526

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered docs/react/customize-theme.en-US.md](https://github.com/ant-design/ant-design/blob/fix/docs-theme/docs/react/customize-theme.en-US.md)
[View rendered docs/react/customize-theme.zh-CN.md](https://github.com/ant-design/ant-design/blob/fix/docs-theme/docs/react/customize-theme.zh-CN.md)

### CODE REVIEW

File: docs/react/customize-theme.en-US.md [hunk 1]
```diff
@@ -198,6 +199,7 @@ module.exports = {
 +     options: {
 +       modifyVars: {
 +          'hack': `true;@import "${require.resolve('antd/lib/style/color/colorPalette.less')}";`,
++          ...defaultTheme, // darkTheme and compactTheme depend on defaultTheme
```
Comment (afc163) [inline docs/react/customize-theme.en-US.md:202]: 应该不需要依赖 `defaultTheme`。

@ycjcl868 看看。

Comment (AshoneA) [reply docs/react/customize-theme.en-US.md:202]: https://github.com/ant-design/ant-design/pull/22934
这里我把 default 从 theme 中抽取出来了。

Comment (afc163) [reply docs/react/customize-theme.en-US.md:202]: 那 umi 插件那里也要改。

Comment (AshoneA) [reply docs/react/customize-theme.en-US.md:202]: https://github.com/umijs/plugins/pull/153
提了

Comment (ycjcl868) [reply docs/react/customize-theme.en-US.md:202]: 抽离后，就需要加默认 theme 了。

**Pull request merged**